src/LeoNerdEncoder.cpp

```diff
@@ -51,15 +51,16 @@ LeoNerdEncoder::~LeoNerdEncoder() {
     if(_intPin != -1)
         detachInterrupt(_intPin);
     #if !defined(__ESP32__) && !defined(ESP8266)
-    Wire.end();
+    if (_I2CBus != nullptr) _I2CBus->end();
     #endif
 }
 
 /**
  * Initialize the encoder library
  */
-void LeoNerdEncoder::begin() {
-    Wire.begin();
+void LeoNerdEncoder::internalBegin(I2CBusBase* bus) {
+    _I2CBus = bus;
+    if (bus != nullptr) _I2CBus->begin();
```

Comment:
If no BusBase is set, return immediately from constructor. Since the class instantiation is invalid, no need to process further

Reply:
Done